### PR TITLE
Using pytest for all Integration Tests

### PR DESCRIPTION
**Pull request by amihaiemil**
PR for #184 

* moved IT for poster authentication under ``tests/integration`` and renamed the file so pytest can see it
* renamed ``Authenticated.token()`` to ``Authenticated.fetch_token()`` since the initial name was conflicting with the encapsulated ``token`` attribute (the test wasn't working as expected)
* Skipped the test since the actual authentication is not yet implemented

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#189](https://github.com/timelesslounge/timelessis/issues/189) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#189](https://github.com/timelesslounge/timelessis/issues/189) is assigned to @vladarefiev[/z](https://www.0crat.com/u/vladarefiev), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/76126887-ae12-4e2d-b497-02dec55ef4dd); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: tests/integration/it_poster_test.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+"""Integration tests for Poster"""
+
+
+@pytest.mark.skip(
+    reason="Authentication mechanism (Authenticated#auth()) is not yet implemented!"
+)
+def test_auth():
+    assert Authenticated(
+        clientid="$0m3C1i3ntId"
+    ).fetch_token() != "", "Poster did not authenticate the user!"
```
Comment (vladarefiev) [inline tests/integration/it_poster_test.py]: @amihaiemil `assert not Authenticated().fetch_token`

**Comment by vladarefiev** [comment]
@amihaiemil good, how should i find here 3 issues :D

File: timeless/poster/api.py [hunk 2]
```diff
@@ -67,4 +67,4 @@ def auth(self):
          implementing uncomment test of poster auth in it_test_poster.py
 
         """
-        raise("poster.auth not implemented yet")
+        raise Exception("poster.auth not implemented yet")
```
Comment (vladarefiev) [inline timeless/poster/api.py]: @amihaiemil mb `NotImplementedError`, WDYT?

**Comment by amihaiemil** [comment]
@vladarefiev changes made, please check. One mention though, the assertion has to be positive, not negative (without ``not``), since we want to assert that the String is contains something, it is not empty.

**Comment by vladarefiev** [comment]
@amihaiemil yep, got it.

**Comment by vladarefiev** [comment]
@rultor merge

File: tests/integration/it_poster_test.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import unittest
+
```
Comment (emilianodellacasa) [inline tests/integration/it_poster_test.py]: @amihaiemil why do we still need this?

**Comment by emilianodellacasa** [comment]
@amihaiemil just a comment

**Comment by amihaiemil** [comment]
@emilianodellacasa fixed, sorry

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @vladarefiev[/z](https://www.0crat.com/u/vladarefiev), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#189](https://github.com/timelesslounge/timelessis/issues/189) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

**Comment by ypshenychka** [comment]
@vladarefiev According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only two issues were found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by vladarefiev** [comment]
@ypshenychka confirm

**Comment by ypshenychka** [comment]
@vladarefiev thanks

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @vladarefiev[/z](https://www.0crat.com/u/vladarefiev)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Comment by krzyk** [comment]
@amihaiemil actually I see you forgot to move the todo to the new file

**Comment by amihaiemil** [comment]
@krzyk maybe, I had a merge conflict with that file and wasn't sure which was the good one. But I think @emilianodellacasa put it back, it's [here](https://github.com/timelesslounge/timelessis/blob/16b132814356b8a14f49c710821eb17a4eab26b8/tests/integration/it_poster_test.py#L8-L10) and actually the task got assigned to me :)

**Pull request merged**